### DISCUSSION TITLE
where is quarto class 'learn-more' coming from?

### DISCUSSION THREAD

**Discussion by ligea-alexander** [quarto-dev/quarto-cli#11582] [Q&A] [answered]
### Description

Hi, just learning through dissecting quarto-web GitHub....I cannot for the life of me figure out where the {.learn-more} class is coming from. It is not in the styles.css or index.css or even theme.css file. It isn't native to Bootstrap so where?

**Comment by mcanouil** [comment]
Could you be a bit more specific? What are you talking about? Where did you see that?

**Reply by ligea-alexander** [reply]
It's here: >> https://github.com/quarto-dev/quarto-web/blob/main/index.qmd

but I don't see it in the styles.css. How is it being referenced? For context, I've found Quarto a bit of a learning curve since in the examples it is not clear that sometimes functionalities we see through Quarto is relying on native Bootstrap and thereby referencing Bootstrap css rather than 'Quarto' stuff.

<img width="424" alt="Screenshot 2024-11-30 at 7 08 30 pm" src="https://github.com/user-attachments/assets/5b84e428-aed9-4e83-a2f6-12bbce01051f">
<img width="220" alt="Screenshot 2024-11-30 at 7 08 42 pm" src="https://github.com/user-attachments/assets/4dd2e088-81f0-4a64-9d78-33fb996618ce">

**Reply by mcanouil** [reply]
I don't understand the purpose of the question. A CSS class does not require to have attributes.
It can be defined simply by using it to for instance allow JavaScript targeting.

Quarto embeds/uses Bootstrap but in the end the HTML format is HTML so you can use whatever web technologies you like.

**Reply by ligea-alexander** [reply]
Thank you for the response. I understand that a CSS class doesn’t require attributes to exist. However, I am trying to trace the .learn-more class specifically within the Quarto codebase to understand its intended functionality or origin. Is it being defined somewhere explicitly in the Quarto source files (e.g., CSS, JavaScript, or Markdown templates), or is it a convention that Quarto users are expected to define on their own?

**Reply by mcanouil** [reply]
It's defined where you took your screenshot and is for use on the documentation.
That's not a general feature, it's a custom thing for the documentation site as any custom thing you can find on various website including mine.

It's just a class used to identify the div, it could have been an ID.
You can see that it as no attributes attached to it:

<img width="1624" alt="image" src="https://github.com/user-attachments/assets/3b21847c-043c-4aeb-89d9-b70afcd9009a">


> Is it being defined somewhere explicitly in the Quarto source files (e.g., CSS, JavaScript, or Markdown templates), or is it a convention that Quarto users are expected to define on their own?

This has nothing to do with Quarto. It's HTML/CSS development.
Sorry, but it remains quite elusive what you are trying to learn or why.

That's simple/standard web development and usage of the fenced div syntax described in https://quarto.org/docs/authoring/markdown-basics.html#sec-divs-and-spans (and in Pandoc documentation).

If you want/need to have an HTML div for whatever reasons (display, etc.), then:

````
::: {.my-whatever-css-class-to-identify-this}
This content will be in an HTML div with the class `my-whatever-css-class-to-identify-this`.
This might be used for later CSS theming, JavaScript targetting or not at all.
:::
````

**Reply by ligea-alexander** [reply]
Thank you for the detailed explanation and example! Yes I understand that .learn-more is specific to the documentation site and not a general Quarto feature. However, my main question remains: Where is this .learn-more class defined in the source files? For example, is it in a CSS file, a JavaScript function, or hardcoded into the Markdown for the documentation? Understanding its origin will help me grasp how similar customizations might work. Thank you for your patience as I’m learning!

**Reply by mcanouil** [reply]
It's defined in the Quarto document by the code you took a screenshot of.
What's not clear about this?

If I write the following in a Quarto document:

```
::: {.my-class}
::: 
```

It means I have defined `.my-class` and I can add attributes via CSS

```css
.my-class {
  ...
}
```

I can write JavaScript to target that class.

Or I can do nothing with it and use it only to "label" my div.

The same is for HTML.
If I write `<div class="my-class"></div` then "my-class" exists thus is defined.


Same things apply to ID:

```
::: {#my-id}
::: 
```

Then "my-id" is defined as an ID, so I can use it to apply CSS rule but I don't need or have to.

```css
#my-id {
  ...
}
```

**Reply by ligea-alexander** [reply, marked as answer]
Ah, okay! It's just a semantic marker or potential hook?

**Reply by mcanouil** [reply]
Yes, sort of speak for this particular example.
